modules/xmpp/FileSharing.ts

```diff
@@ -0,0 +1,108 @@
+import { XMPPEvents } from '../../service/xmpp/XMPPEvents';
+
+import ChatRoom from './ChatRoom';
+import XMPP from './xmpp';
+
+export const IDENTITY_TYPE = 'file-sharing';
+
+/**
+ * The FileSharing logic.
+ */
+export default class FileSharing {
+    private _mainRoom: ChatRoom;
+    private _xmpp: XMPP;
+
+    /**
+     * Constructs File sharing manager for a room.
+     *
+     * @param {ChatRoom} room the main room.
+     */
+    constructor(room: ChatRoom) {
+        this._mainRoom = room;
+        this._xmpp = room.xmpp;
+
+        this._handleMessages = this._handleMessages.bind(this);
+        this._mainRoom.xmpp.addListener(XMPPEvents.FILE_SHARING_EVENT, this._handleMessages);
+    }
+
+    /**
+     * Stops listening for events.
+     */
+    dispose() {
+        this._mainRoom.xmpp.removeListener(XMPPEvents.FILE_SHARING_EVENT, this._handleMessages);
+    }
+
+    /**
+     * Whether AV moderation is supported on backend.
+     *
+     * @returns {boolean} whether AV moderation is supported on backend.
+     */
+    isSupported() {
+        return Boolean(this._xmpp.fileSharingComponentAddress);
+    }
+
+    /**
+     * Returns the file sharing identity type (service name).
+     *
+     * @returns {string} the file sharing service name.
+     */
+    getIdentityType() {
+        return IDENTITY_TYPE;
+    }
+
+    addFile(metadata: any) {
```

Comment:
Pl document the type since this becomes part of the API.

Reply:
Can you document the metadata with a TS interface? This way it will bubble up to JM.